qmlti/LTI_Data_Connector_qmp.php

```diff
@@ -504,6 +503,117 @@ public function Resource_Link_Share_Key_delete($share_key) {
 
   }
 
+###
+###  Result methods
+###
+
+###
+#    Saves the current result into the Results table.
+###
+  public function Results_save($outcome, $consumer, $resource_link, $participant) {
+
+    $time = time();
+    $now = date('Y-m-d H:i:s', $time);
+    $id = $resource_link->getId();
+
+    $sql = 'INSERT INTO ' . $this->dbTableNamePrefix . LTI_Data_Connector::RESULTS_TABLE_NAME . ' (context_id, ' .
+             'assessment_id, customer_id, created, score, result_id) ' .
+             'VALUES (:context, :assessment, :customer, :created, :score, :result)';
+    $query = $this->db->prepare($sql);
+    $query->bindValue('context', $id, PDO::PARAM_STR);
+    $query->bindValue('assessment', $resource_link->getSetting('qmp_assessment_id'), PDO::PARAM_STR);
+    $query->bindValue('customer', $participant, PDO::PARAM_STR);
```

Comment:
I agree! I think the best course of action is to implement a connector via consumer_key, as it is essentially the same as the customer_id, is already used in many of the tables, and can reference a tool consumer which can provide the same security as customer_id.